pkg/model/coordinator/coordinator.go

```diff
@@ -324,11 +325,15 @@ func (coo *Coordinator) InitState(bootstrap bool, startIndex milestone.Index) er
 // Returns non-critical and critical errors.
 func (coo *Coordinator) createAndSendMilestone(parents hornet.MessageIDs, newMilestoneIndex milestone.Index) error {
 
-	messagesMemcache := storage.NewMessagesMemcache(coo.storage)
-	metadataMemcache := storage.NewMetadataMemcache(coo.storage)
+	parents = parents.RemoveDupsAndSortByLexicalOrder()
+
+	messagesMemcache := storage.NewMessagesMemcache(coo.storage.CachedMessage)
+	metadataMemcache := storage.NewMetadataMemcache(coo.storage.CachedMessageMetadata)
+	memcachedParentsTraverserStorage := dag.NewMemcachedParentsTraverserStorage(coo.storage, metadataMemcache)
```

Comment:
Because we are reusing the same `metadataMemcache` sometimes for other stuff, not only the `Traverser`.

Should I make it a variadic parameter?
IMO it obfuscates the cleanup logic, and not sure if that is more readable in the end :shrug: .

Reply:
I had a look at least in the PR and did not see an obvious external usage of the caches. But it is ok if you want to leave it like this